controllers/subscription.go

```diff
@@ -70,30 +72,76 @@ func newSubscription(p api.Pattern) *operatorv1alpha1.Subscription {
 		},
 	}
 
-	if p.Spec.GitOpsConfig.UseCSV {
-		spec.StartingCSV = p.Spec.GitOpsConfig.OperatorCSV
+	if input.Channel != "" {
+		spec.Channel = input.Channel
+	}
+
+	if input.CatalogSource != "" {
+		spec.CatalogSource = input.CatalogSource
+	}
+
+	if input.StartingCSV != "" {
+		spec.StartingCSV = input.StartingCSV
 	}
 
-	if p.Spec.GitOpsConfig.ManualApproval {
-		spec.InstallPlanApproval = operatorv1alpha1.ApprovalManual
+	if spec.InstallPlanApproval != "" {
+		spec.InstallPlanApproval = input.InstallPlanApproval
 	}
 
 	return &operatorv1alpha1.Subscription{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "openshift-gitops-operator",
+			Name:      gitopsSubscriptionName,
 			Namespace: subscriptionNamespace,
 		},
 		Spec: spec,
 	}
 }
 
-func getSubscription(client olmclient.Interface, name, namespace string) (error, *operatorv1alpha1.Subscription) {
+func newSubscriptionFromConfigMap(r kubernetes.Interface) (operatorv1alpha1.Subscription, error) {
+	var newSpec *operatorv1alpha1.Subscription
+
+	// Check if the config map exists and read the config map values
+	if cm, err := r.CoreV1().ConfigMaps(common.OperatorNamespace).Get(context.Background(), common.OperatorConfigFile, metav1.GetOptions{}); err != nil {
+		fmt.Println("Patterns Config Map not found. Using default subscriptions values for OpenShift GitOps.")
+		spec := operatorv1alpha1.SubscriptionSpec{}
+		newSpec = newSubscription(&spec)
+	} else {
+		// Config Map exists
+		// Read config parameters
+
+		common.PatternsOperatorConfig = cm.Data
+
+		var installPlanApproval operatorv1alpha1.Approval
+
+		if configValueWithDefault(common.PatternsOperatorConfig, "gitops.installApprovalPlan", common.GitOpsDefaultApprovalPlan) == "Manual" {
+			installPlanApproval = operatorv1alpha1.ApprovalManual
+		} else {
+			installPlanApproval = operatorv1alpha1.ApprovalAutomatic
+		}
+
+		configSpec := operatorv1alpha1.SubscriptionSpec{
+			CatalogSource:          configValueWithDefault(common.PatternsOperatorConfig, "gitops-source", common.GitOpsDefaultCatalogSource),
```

Comment:
any reason this is not gitops.catalogSource?

Reply:
Typo. Done.